customTests/azure_gpu_bandwidth.nhc

```diff
@@ -173,8 +169,187 @@ function check_gpu_bw()
  
     return 0
 }
 
+function get_OAM_pairs(){
+    # Get OAM pairs
+    # return array where the index is the GPU id and the value is the OAM pair
+    oam_check=$(rocm-smi --showserial | grep GPU)
+
+    if [[ "$oam_check" == *"Not supported"* ]]; then
+        echo ""
+        return 0
+    fi
+
+    mapfile -t gsn < <(rocm-smi --showserial | grep GPU | awk '{print $5}')
+
+    oam_pairs=()
+
+    for ((i=0; i<${#gsn[@]}; i++)); do
+        for ((j=0; j<${#gsn[@]}; j++))
+        do
+            if [[ $i -eq $j ]]; then
+                continue
+            else
+                if [[ "${gsn[$j]}" == *"${gsn[$i]}"* ]]; then
+                    oam_pairs+=($j)
+                fi
+            fi
+        done
+    done
+    echo "${oam_pairs[@]}"
+    # check if the array is empty
+    if [ ${#oam_pairs[@]} -eq 0 ]; then
+        echo ""
+    fi
+    return 0
+}
+
+function check_rocm_gpu_bw(){
+    # Check GPU BW using ROCM Bandwidth Test 
+    exp_hd_bw=$1
+    exp_dd_bw=$2
+    exp_oam_bw=$3
+
+    if [ ! -f /opt/rocm/bin/rocm-bandwidth-test ]; then
+        die 1 "check_gpu_bw: rocm_bandwidth_test not found. Ensure ROCM is installed and sourced correctly."
+        return 1
+    fi
+
+    cmd="sudo /opt/rocm/bin/rocm-bandwidth-test -a"
+    err_msg="check_gpu_bw: Failed to run rocm-bandwidth-test with error code"
+    if ! catch_error "$cmd" "$error_nbw"; then
+        return 1
+    fi
+    rocm_result=$output
+    lines=()
+    IFS=$'\n'
+    lines=( $rocm_result)
+    IFS=$' \t\n'
+
+    # Get OAM pairs
+    oam_pairs=$(get_OAM_pairs)
+    if [ -z "$oam_pairs" ]; then
+        oam_pairs=()
+    fi
+    read -a oam_pairs <<< "$oam_pairs"
+
+    device_results=()
+    declare -A devTypeIDmap # used to map a device to a type: CPU or GPU
+    CPUNum=$(lscpu | grep "Socket(s)" | awk '{print $2}')
+    store=false
+    for line in "${lines[@]}"; do
+        if [[ "$line" == *"Device: "* ]]; then
+            id=$( echo $line | awk  -F ',' '{print $1}' | awk '{print $2}' )
+            if [[ "$line" == *"GPU"* ]]; then
+                devTypeIDmap[$id]="GPU"
+            else
+                devTypeIDmap[$id]="CPU"
+            fi
+        fi
+        if [[ "$line" == *"Unidirectional copy"* ]]; then
+            store=true
+            continue
+        fi
+        if [ "$store" == true ]; then
+            device_results+=("$line")
+        fi
+    done
+
+    bad_results=()
+    for ((i=1; i<${#device_results[*]}; i++))
+    do  
+        val_list=()
+        values="${device_results[$i]}"
+        IFS=$' ' read -ra values <<< "$values"
+        id=${values[0]}
+
+        for ((devid=0; devid<$(( ${#values[*]} -1 )); devid++)) 
+        do
+            val=${values[$(( devid + 1 ))]} #offset first val
+            real_GPUID=$(( devid - CPUNum )) # offset by number of CPUs
+            cur_real_GPUID=$(( id - CPUNum )) # offset by number of CPUs for GPU being evaluated
+            dbg "Rocm Device id $id to Device id $devid: $val GB/s"
```

Comment:
ROCm

Reply:
I'll go through all docs and make sure its right

Reply:
addressed